test/common/alerts_mechanism.go

```diff
@@ -31,73 +26,79 @@ type alertManagerConfig struct {
-	monitoringTimeout                         = time.Minute * 25
-	monitoringRetryInterval                   = time.Minute
-	verifyOperatorDeploymentTimeout           = time.Minute * 5
-	verifyOperatorDeploymentRetryInterval     = time.Second * 15
+	keycloakOperatorDeploymentName        = "keycloak-operator"
+	monitoringTimeout                     = time.Minute * 20
+	monitoringRetryInterval               = time.Minute * 1
+	verifyOperatorDeploymentTimeout       = time.Minute * 5
+	verifyOperatorDeploymentRetryInterval = time.Second * 10
 )
 
-var threescaleAlertsToTest = map[string]string{
-	"RHMIThreeScaleApicastProductionServiceEndpointDown": "none",
-	"ThreeScaleApicastProductionPod":                     "none",
+var keycloakAlertsToTest = map[string]string{
+	"RHMIUserRhssoKeycloakOperatorMetricsServiceEndpointDown": "none",
 }
 
 // TestIntegreatlyAlertsMechanism verifies that alert mechanism works
 func TestIntegreatlyAlertsMechanism(t TestingTB, ctx *TestingContext) {
 
-	originalOperatorReplicas, err := getNumOfReplicasDeployment(threescaleOperatorDeploymentName, ctx.KubeClient)
+	originalOperatorReplicas, err := getNumOfReplicasDeployment(keycloakOperatorDeploymentName, ctx.KubeClient)
 	if err != nil {
 		t.Errorf("failed to get number of replicas: %s", err)
 	}
 
 	// verify that alert to be tested is not firing before starting the test
-	err = getThreescaleAlertState(ctx)
+	err = getKeycloakAlertState(ctx)
 	if err != nil {
 		t.Fatal("failed to get threescale alert state", err)
 	}
 
-	threescaleAlertsFiring := false
+	keycloakAlertsFiring := false
 
-	for threescaleAlertName, threescaleAlertState := range threescaleAlertsToTest {
-		if threescaleAlertState != "none" {
-			threescaleAlertsFiring = true
-			t.Errorf("%s alert should not be firing", threescaleAlertName)
+	for keycloakAlertName, keycloakAlertState := range keycloakAlertsToTest {
+		if keycloakAlertState != "none" {
+			keycloakAlertsFiring = true
+			t.Errorf("%s alert should not be firing", keycloakAlertName)
 		}
 	}
 
-	if threescaleAlertsFiring {
-		t.Log("3Scale alerts firing unexpectedly")
+	if keycloakAlertsFiring {
+		t.Log("Keycloak alerts firing unexpectedly")
 		t.FailNow()
 	}
 
-	// scale down Threescale operator and product pods and verify that threescale alert is firing
+	t.Log("Scaling down keycloak operator deployment")
+	scaleDeployment(t, keycloakOperatorDeploymentName, 0, ctx.KubeClient)
+
+	t.Log("Keycloak alerts are not firing - performing tests")
 	err = performTest(t, ctx, originalOperatorReplicas)
 	if err != nil {
-		t.Log("Error During scale down 3Scale operator and product pods and verify that 3scale alert is firing")
+		t.Log("Error During scale down keycloak operator and product pods and verify that keycloak alert is firing")
 		t.Fatal(err)
 	}
 
+	t.Log("Scaling up keycloak operator deployment")
+	scaleDeployment(t, keycloakOperatorDeploymentName, originalOperatorReplicas, ctx.KubeClient)
```

Comment:
This function needs to happen even if the test fails at an earlier stage such as the error check on line 73. Please add this as a deferred function.